wikimedia/wikipedia-ios
# Migrate MWKLanguageLinkController to Core Data

**Phabricator:** https://phabricator.wikimedia.org/T257250 

### Notes

When `UserDefaults` were [migrated off the named suite](https://github.com/wikimedia/wikipedia-ios/pull/3439) into using the standard suite, the today widgets stopped accurately polling the currently selected app language. This was because the today widgets were referencing the standard `UserDefaults` suite, which isn't shared between the main app and the extensions. Joe migrated the app language component here from `UserDefaults` into Core Data, which should mitigate the issue.

I'll review this one myself because it relates to some widget stuff I'm working on. (It won't let me self-assign for review, so I'll self-assign as an assignee).

### Test Steps
1. Open app, change primary language
2. Open today widgets (e.g. Top read), confirm the widget's language matches the primary language selected in app
